Backpack/Price/Classes/BPKPrice.swift

```diff
@@ -0,0 +1,203 @@
+    }()
+    
+    private lazy var leadingTextLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: footnoteOrCaptionFontStyle())
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
+    
+    private lazy var topTextStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [lineThroughTextLabel, separatorLabel, leadingTextLabel])
+        stackView.axis = .horizontal
+        stackView.spacing = BPKSpacingSm
+        return stackView
+    }()
+    
+    private lazy var priceStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [priceLabel, trailingTextLabel])
+        stackView.axis = .horizontal
+        stackView.spacing = BPKSpacingSm
+        stackView.alignment = .firstBaseline
+        return stackView
+    }()
+    
+    private lazy var containerStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [topTextStackView, priceStackView])
+        stackView.axis = .vertical
+        return stackView
+    }()
+    
+    public init(
+        price: String,
+        leadingText: String? = nil,
+        lineThroughText: String? = nil,
+        trailingText: String? = nil,
+        alignment: Alignment = .left,
+        size: Size = .large
+    ) {
+        self.price = price
+        self.leadingText = leadingText
+        self.lineThroughText = lineThroughText
+        self.trailingText = trailingText
+        self.alignment = alignment
+        self.size = size
+        
+        super.init(frame: .zero)
+        
+        leadingTextLabel.text = leadingText
+        separatorLabel.text = "•"
+        priceLabel.text = price
+        trailingTextLabel.text = trailingText
+        
+        lineThroughTextLabel.text = lineThroughText
+        applyLineThroughStyling()
+        
+        setupView()
+        setupConstraints()
+    }
+    
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func setupView() {
+        [
+            priceStackView,
+            containerStackView
+        ].forEach {
+            $0.translatesAutoresizingMaskIntoConstraints = false
+        }
+        
+        lineThroughTextLabel.isHidden = lineThroughText == nil
+        leadingTextLabel.isHidden = leadingText == nil
+        trailingTextLabel.isHidden = trailingText == nil
+        separatorLabel.isHidden = lineThroughTextLabel.isHidden || leadingTextLabel.isHidden
+        
+        switch alignment {
+        case .left:
+            containerStackView.alignment = .leading
+            priceStackView.axis = .horizontal
+            priceStackView.spacing = BPKSpacingSm
+        case .right:
+            containerStackView.alignment = .trailing
+            
+            if size == .small {
+                priceStackView.axis = .vertical
+                priceStackView.spacing = BPKSpacingNone
+            }
+        }
```

Comment:
should this be executed every time the properties are set? ie: if I set trailingText after this got executed, this should be handled as well.
I recommend putting this logic inside a `updateLookAndFeel` method, and calling it from here and from the didSet of the exposed properties